src/app/project/board/attribution-tab-view/attribution-tab-view.component.ts

```diff
@@ -8,15 +8,32 @@ import { IAttribution, IOrganisation, IPerson } from '../dataset-metadata';
 })
 export class AttributionTabViewComponent {
     // attribution input
-    @Input() attributions: IAttribution[];
+    @Input() attributions: Attribution[];
+
+    @Input() subProperties: Object;
+
+    currentAgent: Person | Organization | IId;
 
     // return the type of agent to use correct template to display it
-    getAgentType(agent: IPerson | IOrganisation) {
-        if (agent.hasOwnProperty('familyName')) {
+    setAgent (agent: Person | Organization | IId) {
+        let atype = this.getAgentType(agent);
+        if (atype) {
+            this.currentAgent = agent;
+            return atype;
+        }
+        this.currentAgent = this.subProperties[agent.id];
+        atype = this.getAgentType(this.currentAgent);
+        return atype;
+    }
+
+    getAgentType (agent: Person | Organization | IId) {
+        if (agent instanceof Person) {
             return 'person';
         }
-        if (agent.hasOwnProperty('url')) {
+        if (agent instanceof Organization) {
             return 'organisation';
         }
+        return undefined;
     }
```

Comment:
method return type is missing and `if...else`

Reply:
linter was complaining same for else block here so I removed it.

Reply:
This is strange actually. In some cases `if...else` match perfect, in others `if...else if`. This is not big thing here, so let's leave it right now.

Reply:
I've applied below and on vscode, linter doesn't complain. Could you show me (screen) how it looks on your side?

```
    setAgent (agent: Person | Organization | IId): string {
        let atype = this._datasetMetadataService.getContactType(agent);
        if (atype) {
            this.currentAgent = agent;
        } else {
            this.currentAgent = this.subProperties[agent.id];
            atype = this._datasetMetadataService.getContactType(this.currentAgent);
        }
        return atype;
    }
```

Reply:
Here it is: [https://github.com/dasch-swiss/dsp-app/runs/1987482010?check_suite_focus=true](https://github.com/dasch-swiss/dsp-app/runs/1987482010?check_suite_focus=true.
It shows _misplaced else_ error in multiple places.

Reply:
It's like that because of wrong formatting (see your code in screen below). There shouldn't be ENTER between if and else block. See my example above.

![image](https://user-images.githubusercontent.com/29251288/109626771-39ef3200-7b41-11eb-948f-c66d350b9f2c.png)

Reply:
Thanks for pointing it out. Changes are done in [d8b4993](https://github.com/dasch-swiss/dsp-app/pull/400/commits/d8b4993b6648e3fe469ecdf0016843e8d0b8d78c).